### ISSUE TITLE
Do Post-Flush/Compaction Consistency Test on Encrypted Data

### ISSUE THREAD

**Issue by rahuldesirazu** [yugabyte/yugabyte-db#3456]
After flushing a file and before adding it to the MANIFEST, do a consistency check on the file to make sure the checksums are valid.

**Comment by bmatican** [comment]
@rahuldesirazu is the ldb consistencycheck actually doing pretty much this? opening the files block by block and validating their checksums?

**Comment by rahuldesirazu** [comment]
@bmatican Yes it is, and we should try to integrate it into the flush path to detect data corruption before it gets committed.
